src/components/Wizard/index.tsx

```diff
@@ -2,24 +2,65 @@ import React from 'react';
 import Button from "@material-ui/core/Button";
 import './styles.css';
 import { connect, ConnectedProps } from 'react-redux';
-import { RootState, useSelector } from "../../reducers/rootReducer";
+import { RootState } from "../../reducers/rootReducer";
 import { 
   setCurrentStep as setCurrentStepAction,
-  setStepsCompleted as setStepsCompletedAction
+  setStepsCompleted as setStepsCompletedAction,
+  SetCurrentStep,
+  SetStepsCompleted
 } from "../../actions/docSteps";
 
+interface PrevBtn {
+  currentStep: number,
+  setCurrentStep: (step: number) => SetCurrentStep,
+}
+
+interface NextBtn {
+  currentStep: number,
+  setCurrentStep: (step: number) => SetCurrentStep,
+  totalSteps: number
+}
+
+interface FinishBtn {
+  currentStep: number,
+  setStepsCompleted: () => SetStepsCompleted,
+  totalSteps: number,
+}
+
+const PreviousButton: React.FC<PrevBtn> = ({ currentStep, setCurrentStep }) => {
+  if (currentStep <= 0) return null;
+  return (
+    <Button onClick={() => setCurrentStep(currentStep - 1)}>Previous</Button>
+  );
+}
+
+const NextButton: React.FC<NextBtn> = ({ currentStep, setCurrentStep, totalSteps }) => {
+  if (currentStep >= totalSteps - 1) return null;
+  return (
+    <Button onClick={() => setCurrentStep(currentStep + 1)}>Next</Button>
+  );
+}
+
+const FinishButton: React.FC<FinishBtn> = ({ currentStep, setStepsCompleted, totalSteps }) => {
+  if (currentStep !== totalSteps - 1) return null;
+  return (
+    <Button onClick={setStepsCompleted}>Finish</Button>
+  );
+}
+
 const Wizard: React.FC<Props> = (props) => {
   const { children, docSteps, setCurrentStep, setStepsCompleted } = props;
-  const currentStep = useSelector(state => state.docSteps.currentStep) || 0;
-  const steps = useSelector(state => state.docSteps.steps);
+  const currentStep = docSteps.currentStep || 0;
+  const steps = docSteps.steps;
+  const totalSteps = docSteps.steps.length;
   return (
     <div className="wizard">
       {children}
       {steps && (
         <div className="action-buttons">
-          {currentStep > 0 && <Button onClick={() => setCurrentStep(currentStep - 1)}>Previous</Button>}
-          {currentStep < docSteps.steps.length && <Button onClick={() => setCurrentStep(currentStep + 1)}>Next</Button>}
-          {currentStep === docSteps.steps.length && <Button onClick={setStepsCompleted}>Finish</Button>}
+          <PreviousButton currentStep={currentStep} setCurrentStep={setCurrentStep} />
```

Comment:
How do you feel about this ? , I really like it.

Reply:
Yeah, it's easier to read, and responsibilities are better delegated